odoo/addons/base/models/res_users.py

```diff
@@ -176,7 +172,9 @@ def _search_full_name(self, operator, operand):
             group_name = values.pop().strip()
             category_name = values and '/'.join(values).strip() or group_name
             group_domain = [('name', operator, lst and [group_name] or group_name)]
-            category_domain = [('category_id.name', operator, lst and [category_name] or category_name)]
+            category_ids = self.env['ir.module.category'].sudo().search(
+                [('name', operator, lst and [category_name] or category_name)]).ids
```

Comment:
You can inject this one as a subquery with
```suggestion
            category_ids = self.env['ir.module.category'].sudo()._search(
                [('name', operator, [category_name] if lst else category_name)])
```